`int-conversion` build failure with newer clang

seeing some int-conversion build errors with newer clang as below

```
getfcomment/main.c:152:47: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'FSCatalogInfoBitmap' (aka 'unsigned int') [-Wint-conversion]
    152 |             err = hfsdata/main.c:599:38: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'FSCatalogInfoBitmap' (aka 'unsigned int') [-Wint-conversion]
    FSGetCatalogInfo (&fileRef, NULL, NULL, NULL, &fileSpec, NULL);
        |                                               ^~~~
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/__stddef_null.h599 |     err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
        |                                      ^~~~
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/__stddef_null.h:26:14: note: expanded from macro 'NULL'
     26 | #define NULL ((void *)0)
        |              ^~~~~~~~~~~
  :26:14: note: expanded from macro 'NULL'
     26 | #define NULL ((void *)0)
        |              ^~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:2620:70: note: passing argument to parameter 'whichInfo' here
  /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:2620:70: note: passing argument to parameter 'whichInfo' here
   2620 | extern OSErr 2620 | extern OSErr  FSGetCatalogInfo(const FSRef *ref, FSCatalogInfoBitmap whichInfo, FSCatalogInfo *catalogInfo, HFSUniStr255 *outName, FSSpecPtr fsSpec, FSRef *parentRef) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
        |                                                                      ^
    FSGetCatalogInfo(const FSRef *ref, FSCatalogInfoBitmap whichInfo, FSCatalogInfo *catalogInfo, HFSUniStr255 *outName, FSSpecPtr fsSpec, FSRef *parentRef) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
        |                                                                      ^
  hfsdata/main.c:639:38: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'FSCatalogInfoBitmap' (aka 'unsigned int') [-Wint-conversion]
    639 |     err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
        |                                      ^~~~
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/__stddef_null.h:26:14: note: expanded from macro 'NULL'
     26 | #define NULL ((void *)0)
        |              ^~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:2620:70: note: passing argument to parameter 'whichInfo' here
   2620 | extern OSErr  FSGetCatalogInfo(const FSRef *ref, FSCatalogInfoBitmap whichInfo, FSCatalogInfo *catalogInfo, HFSUniStr255 *outName, FSSpecPtr fsSpec, FSRef *parentRef) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
        |                                                                      ^
  hfsdata/main.c:733:38: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'FSCatalogInfoBitmap' (aka 'unsigned int') [-Wint-conversion]
    733 |     err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
        |                                      ^~~~
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/__stddef_null.h:26:14: note: expanded from macro 'NULL'
     26 | #define NULL ((void *)0)
        |              ^~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:2620:70: note: passing argument to parameter 'whichInfo' here
   2620 | extern OSErr  FSGetCatalogInfo(const FSRef *ref, FSCatalogInfoBitmap whichInfo, FSCatalogInfo *catalogInfo, HFSUniStr255 *outName, FSSpecPtr fsSpec, FSRef *parentRef) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
        |                                                                      ^
  hfsdata/main.c:775:38: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'FSCatalogInfoBitmap' (aka 'unsigned int') [-Wint-conversion]
    775 |     err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
        |                                      ^~~~
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/__stddef_null.h:26:14: note: expanded from macro 'NULL'
     26 | #define NULL ((void *)0)
        |              ^~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:2620:70: note: passing argument to parameter 'whichInfo' here
   2620 | extern OSErr  FSGetCatalogInfo(const FSRef *ref, FSCatalogInfoBitmap whichInfo, FSCatalogInfo *catalogInfo, HFSUniStr255 *outName, FSSpecPtr fsSpec, FSRef *parentRef) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
        |                                                                      ^
  1 error generated.
  hfsdata/main.c:[103](https://github.com/Homebrew/homebrew-core/actions/runs/10798684389/job/29952719740#step:5:104)6:46: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'FSCatalogInfoBitmap' (aka 'unsigned int') [-Wint-conversion]
   1036 |             err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
        |                                              ^~~~
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/__stddef_null.h:26:14: note: expanded from macro 'NULL'
     26 | #define NULL ((void *)0)
        |              ^~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:2620:70: note: passing argument to parameter 'whichInfo' here
   2620 | extern OSErr  FSGetCatalogInfo(const FSRef *ref, FSCatalogInfoBitmap whichInfo, FSCatalogInfo *catalogInfo, HFSUniStr255 *outName, FSSpecPtr fsSpec, FSRef *parentRef) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
        |                                                                      ^
  5 errors generated.
```

relates to https://github.com/Homebrew/homebrew-core/pull/185193

--- a/getfcomment/main.c
+++ b/getfcomment/main.c
@@ -149,7 +149,7 @@ static void PrintOSXComment (char *path)
             }
 
             //retrieve filespec from file ref
-            err = FSGetCatalogInfo (&fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+            err = FSGetCatalogInfo (&fileRef, 0, NULL, NULL, &fileSpec, NULL);
             if (err != noErr)
             {
 				fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec for %s\n", err, path);

--- a/geticon/main.m
+++ b/geticon/main.m
@@ -144,7 +144,7 @@ static char* CutSuffix (char *name)
   {
     if (name[len-i] == '.')
     {
-      name[len-i] = NULL;
+      name[len-i] = '\0';
       return name;
     }
   }

--- a/hfsdata/main.c
+++ b/hfsdata/main.c
@@ -596,7 +596,7 @@ static OSErr PrintFileType (FSRef *fileRef)
 	}
 
 	// retrieve filespec from file ref
-    err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+    err = FSGetCatalogInfo (fileRef, 0, NULL, NULL, &fileSpec, NULL);
     if (err != noErr)
     {
         fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec from file reference\n", err);
@@ -636,7 +636,7 @@ static OSErr PrintCreatorCode (FSRef *fileRef)
 	}
 
 	// retrieve filespec from file ref
-    err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+    err = FSGetCatalogInfo (fileRef, 0, NULL, NULL, &fileSpec, NULL);
     if (err != noErr)
     {
         fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec from file reference\n", err);
@@ -730,7 +730,7 @@ static OSErr PrintLabelName (FSRef *fileRef)
 	int			labelNum = 0;
 
 	/* retrieve filespec from file ref */
-    err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+    err = FSGetCatalogInfo (fileRef, 0, NULL, NULL, &fileSpec, NULL);
     if (err != noErr)
     {
         fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec from file reference\n", err);
@@ -772,7 +772,7 @@ static OSErr PrintLabelNumber (FSRef *fileRef)
 	int			labelNum = 0;
 
 	/* retrieve filespec from file ref */
-    err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+    err = FSGetCatalogInfo (fileRef, 0, NULL, NULL, &fileSpec, NULL);
     if (err != noErr)
     {
         fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec from file reference\n", err);
@@ -1033,7 +1033,7 @@ static OSErr PrintOSXComment (FSRef	*fileRef)
 	AEIdleUPP inIdleProc = NewAEIdleUPP(&MyAEIdleCallback);
 
             //retrieve filespec from file ref
-            err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+            err = FSGetCatalogInfo (fileRef, 0, NULL, NULL, &fileSpec, NULL);
             if (err != noErr)
             {
 				fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec for %s\n", err);
@@ -1073,7 +1073,7 @@ static OSErr PrintOS9Comment (FSRef *fileRef)
 	char	comment[255] = "\0";
 
 	//retrieve filespec from file ref
-	err = FSGetCatalogInfo (fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+	err = FSGetCatalogInfo (fileRef, 0, NULL, NULL, &fileSpec, NULL);
 	if (err != noErr)
 	{
 		fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec\n", err);

--- a/mkalias/mkalias.c
+++ b/mkalias/mkalias.c
@@ -229,7 +229,7 @@ static void CreateAlias (char *srcPath, char *destPath)
   }
 
   //retrieve source filespec from source file ref
-  err = FSGetCatalogInfo (&srcRef, NULL, NULL, NULL, &sourceFSSpec, NULL);
+  err = FSGetCatalogInfo (&srcRef, 0, NULL, NULL, &sourceFSSpec, NULL);
   if (err != noErr)
   {
     fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec from source FSRef\n", err);

--- a/setfcomment/setfcomment.c
+++ b/setfcomment/setfcomment.c
@@ -166,7 +166,7 @@ static void SetFileComment (char *path, char *comment)
             }
 
             //retrieve filespec from file ref
-            err = FSGetCatalogInfo (&fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+            err = FSGetCatalogInfo (&fileRef, 0, NULL, NULL, &fileSpec, NULL);
             if (err != noErr)
             {
 				fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec for %s\n", err, path);

--- a/setlabel/main.c
+++ b/setlabel/main.c
@@ -165,7 +165,7 @@ static void SetFileLabel (char *path, short labelNum)
   }
 
   //retrieve filespec from file ref
-  err = FSGetCatalogInfo (&fileRef, NULL, NULL, NULL, &fileSpec, NULL);
+  err = FSGetCatalogInfo (&fileRef, 0, NULL, NULL, &fileSpec, NULL);
   if (err != noErr)
   {
     fprintf(stderr, "FSGetCatalogInfo(): Error %d getting file spec for %s\n", err, path);

